### PR TITLE
fix(platform): search-field, fix empty suggestions

### PR DESCRIPTION
**Pull request by Betrozov**
## Related Issue

closes https://github.com/sap/fundamental-ngx/issues/7051

## Description

Fixed error in the console when there is a SearchField without suggestions being passed.

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [n/a] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [n/a] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [n/a] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [n/a] check for console logs (warnings, errors)
-   [n/a] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [n/a] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

##### PR Review

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

### CODE REVIEW

**Comment by droshev** [comment]
@Betrozov can you add an example on the doc page as well?

File: libs/platform/src/lib/search-field/search-field.component.ts [hunk 1]
```diff
@@ -233,53 +209,96 @@ export class SearchFieldComponent extends BaseComponent implements OnInit, OnDes
      * Observable list of string values taken from `suggestions` to populate dropdown menu.
      * @hidden
      */
-    public dropdownValues$: Observable<string[]>;
+    dropdownValues$: Observable<string[]> = of([]);
```
Comment (N1XUS) [inline libs/platform/src/lib/search-field/search-field.component.ts:212]: If it's hidden, shouldn't it be prefixed with _?

File: libs/platform/src/lib/search-field/search-field.component.ts [hunk 2]
```diff
@@ -233,53 +209,96 @@ export class SearchFieldComponent extends BaseComponent implements OnInit, OnDes
...
+    inputId = '';
+
+    /** @hidden */
+    submitId = '';
 
-    public isOpen = false;
-    public isRefresh = false;
-    public isSearchDone = false;
+    /** @hidden */
+    menuId = '';
```
Comment (N1XUS) [inline libs/platform/src/lib/search-field/search-field.component.ts:245]: _ prefix?

**Review by platon-rov — APPROVED** [review]

**Review by N1XUS — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]
Nice!

**Pull request merged**